Automattic/wp-calypso
# A4A: Implement Shopping cart state based on Session storage.

This pull request adds a shopping cart state based on session storage. Storing the cart state on session storage allows the UI to remember previously selected items. It is common practice to use session storage as a caching mechanism for shopping carts on the frontend because it only keeps the items within the browser's tab context.


https://github.com/Automattic/wp-calypso/assets/56598660/c3c4dea5-99cf-402e-894a-56c06e0c2935




Closes https://github.com/Automattic/jetpack-genesis/issues/287

## Proposed Changes
* The PR renamed the Issue License context to the Shopping Cart Context to make the code more readable. The Issue License context no longer matches the Marketplace, which uses the Shopping cart model. Note that the IssueLicense component will be renamed in a separate PR.
* The PR added a new `useShoppingCart` hook, which will contain Shopping cart logic. The hook uses Session storage to cache the state. With session storage, we only cache the state until the page is closed (Browser tab). This is a common practice when implementing a frontend-based Shopping cart.
 

## Testing Instructions
* Switch branch: `git checkout add/a4a/persisting-shopping-cart-state`.
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/marketplace
* Select some items and confirm it is added to the shopping cart.
* Refresh the page to confirm that the shopping cart remembers the selected items.
* Go to the Hosting page and confirm that the shopping cart shows the exact items selected on the Products page.

## Pre-merge Checklist
